File: Shared/src/main/java/com/lenta/shared/models/core/Manufacturer.java
package com.lenta.shared.models.core;

import android.support.annotation.Nullable;

import com.fasterxml.jackson.annotation.JsonProperty;

import lombok.Getter;

public class Manufacturer {

    public static final Manufacturer DEFAULT = new Manufacturer(null, "Нет производителя");

    @JsonProperty("ZPROD")
    @Getter public String code;

    @JsonProperty("PROD_NAME")
    @Getter public String name;

    public Manufacturer(String code, String name)
    {
        this.code = code;
        this.name = name;
    }

    @Override
    public String toString()
    {
        return name;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (obj == null || getClass()!= obj.getClass())
        {
            return false;
        }

        Manufacturer manufacturer =(Manufacturer)obj;
        if (manufacturer == null)
        {
            return false;
        }
        return equals(manufacturer);
    }

    public boolean equals(Manufacturer manufacturer)
    {
        if (manufacturer == null)
        {
            return false;
        }
        return code.equals(manufacturer.code);
    }

    @Override
    public int hashCode()
    {
        return code.hashCode();
    }
}

File: Shared/src/main/java/com/lenta/shared/models/core/Manufacturer.kt
package com.lenta.shared.models.core

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import kotlinx.android.parcel.Parcelize

@Parcelize
data class Manufacturer(
        @SerializedName("ZPROD")
        val code: String,
        @SerializedName("PROD_NAME")
        val name: String
) : Parcelable {

    companion object {
        val DEFAULT = Manufacturer("", "Нет производителя")
    }

    override fun toString(): String {
        return name
    }

    override fun hashCode(): Int {
        return code.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other?.javaClass != javaClass) return false

        other as Manufacturer

        return equals(other)
    }

    fun equals(manufacturer: Manufacturer?): Boolean {
        return if (manufacturer == null) false else code == manufacturer.code
    }
}